lib/garage/restful_actions.rb

```diff
@@ -32,7 +32,22 @@ def resource_class=(klass)
       end
 
       def resource_class
-        @resource_class ||= name.sub(/Controller\z/, '').demodulize.singularize.constantize
+        @resource_class ||= resource_class_name.constantize
+      end
+
+      private
+
+      def resource_class_name
+        class_name = name.sub(/Controller\z/, '').demodulize.singularize
+        class_name_with_suffix = "#{class_name}Resource"
+        case
+        when const_defined?(class_name_with_suffix)
```

Comment:
Does it work under `Rails.env.development?` (or other lazy-loading) environment?